src/foam/u2/search/DateSearchView.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.search',
+  name: 'DateSearchView',
+  extends: 'foam.u2.View',
+
+  documentation: `A SearchView for properties of type Date.`,
+
+  exports: [
+    'as data'
+  ],
+
+  properties: [
+    {
+      name: 'property',
+      documentation: `The property that this view is filtering by. Should be of
+          type Date.`,
+      required: true
+    },
+    {
+      class: 'String',
+      name: 'qualifier',
+      documentation: `Lets the user choose a predicate to filter the view by.`,
+      view: {
+        class: 'foam.u2.view.ChoiceView',
+        choices: [
+          ['True', '--'],
+          ['Gt', 'after'],
+          ['Lt', 'before'],
+        ],
+        defaultValue: 'True'
+      }
+    },
+    {
+      class: 'Date',
+      name: 'date',
+      documentation: `A date selectable by the user. Used as the second argument
+          to the predicate generated by this view.`,
+    },
+    {
+      name: 'predicate',
+      documentation: `All SearchViews must have a predicate as required by the
+          SearchManager. The SearchManager will read this predicate and use it
+          to filter the dao being displayed in the view.`,
+      expression: function(qualifier, date) {
+        if ( qualifier.length > 0 && date && ! isNaN(date.valueOf()) ) {
+          return foam.mlang.predicate[qualifier].create({
+            arg1: this.property,
+            arg2: new Date(date)
+          });
+        }
+        return foam.mlang.predicate.True.create();
+      }
+    },
+    {
+      name: 'name',
+      documentation: `Required by SearchManager.`,
+      value: 'currency search view'
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this
+        .addClass(this.myClass())
+        .start(this.QUALIFIER)
+          .start('div').addClass(this.myClass('carrot')).end()
+        .end()
+        .add(this.DATE);
+    },
+
+    /**
+     * Clears the fields to their default values.
+     * Required on all SearchViews. Called by ReciprocalSearch.
+     */
+    function clear() {
+      this.qualifier = 'True';
+    }
+  ],
+
+  css: `
```

Comment:
I prefer to put css near the top with constants and messages since it's more of a class/static kind of thing. But your choice.

Reply:
I chose to put it at the bottom because CSS usually takes up a lot of space and I rarely need to look at it after it's written the first time. It's at the bottom to avoid having to scroll past it every time I want to add a require or check a property and then go back to the initE function. I'll just leave it at the bottom if it's up to me.

Reply:
ok